consensus/hotstuff/validator/validator.go

```diff
@@ -129,63 +186,72 @@ func (v *Validator) ValidateProposal(proposal *model.Proposal) error {
 		return newInvalidBlockError(block, fmt.Errorf("proposer %s is not leader (%s) for view %d", block.ProposerID, leader, block.View))
 	}
 
-	// check the proposer is a valid committee member at the most recent state on this fork
-	// TODO: move this check into SafetyRules when front-loading the verification
-	_, err = v.committee.IdentityByBlock(block.BlockID, block.ProposerID)
-	if model.IsInvalidSignerError(err) {
-		return newInvalidBlockError(block, fmt.Errorf("proposer %x is leader but is not a valid committee member", block.ProposerID))
+	// The Block must contain a proof that the primary legitimately entered the respective view.
+	// Transitioning to proposal.Block.View is possible either by observing a QC or a TC for the
+	// previous round. If and only if the QC is _not_ for the previous round we require a TC for
+	// the previous view to be present.
+	lastViewSuccessful := proposal.Block.View == proposal.Block.QC.View+1
+	if !lastViewSuccessful {
+		// check if proposal is correctly structured
+		if proposal.LastViewTC == nil {
+			return newInvalidBlockError(block, fmt.Errorf("last view has ended with timeout but proposal doesn't include LastViewTC"))
```

Comment:
These checks happen at the compliance engine, which has no knowledge of what the current view is:

```suggestion
			return newInvalidBlockError(block, fmt.Errorf("QC in block is not for previous view, so expecting a TC but none is included in block"))
```